autobump-maven/src/main/java/com/github/autobump/maven/model/MavenDependencyResolver.java

```diff
@@ -34,20 +33,39 @@ public MavenDependencyResolver() {
 
     @Override
     public Set<Dependency> resolve(Workspace workspace) {
+        return resolve(workspace, new HashSet<>());
+    }
+
+    @Override
+    public Set<Dependency> resolve(Workspace workspace, Set<Dependency> toBeIgnored) {
         Model model = mavenModelAnalyser.getModel(workspace);
+        toBeIgnored.add(Dependency
+                .builder()
+                .name(model.getArtifactId())
+                .group(model.getGroupId())
+                .version(model.getVersion())
+                .build());
         Set<Dependency> dependencies = getDependencies(model);
         dependencies.addAll(getPlugins(model));
         dependencies.addAll(getParentDependency(model));
-        dependencies.addAll(resolveModules(workspace, model.getModules()));
+        dependencies.addAll(resolveModules(workspace, model.getModules(), toBeIgnored));
         dependencies.addAll(getDependenciesFromDependencyManagementSection(model));
-        return dependencies;
+        return dependencies.stream().filter(dependency ->
+                !toBeIgnored.contains(Dependency.builder()
+                        .group(dependency.getGroup())
+                        .version(dependency.getVersion())
+                        .name(dependency.getName())
+                        .build()))
+                .filter(dependency ->
```

Comment:
wouldnt this filter be enough, also Tim's remark

Reply:
the first filter filters out the dependencies that have no version declared the second one filters out all the dependencies the have declared the version as a property and the property couldn't be resolved